x-pack/plugins/apm/public/components/app/transaction_details/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -203,27 +225,39 @@ function getRootTransaction(childrenByParentId: IWaterfallGroup) {
   }
 }
 
-export type IServiceColors = Record<string, string>;
+function getLegends(waterfallItems: IWaterfallItem[]) {
+  const onlySpanItems = waterfallItems.filter(
+    (item) => 'legendValues' in item
+  ) as IWaterfallSpanItem[];
+
+  const legends = [
+    WaterfallLegendType.ServiceName,
+    WaterfallLegendType.SpanType,
+  ].flatMap((legendType) => {
+    const values = uniq(
```

Comment:
Nit: perhaps more clear:

```suggestion
    const legendLabels = uniq(
```

Reply:
changed it to `allLegendValues`